infra/gcp/clusters/projects/k8s-infra-prow-build-trusted/prow-build-trusted/main.tf

```diff
@@ -113,51 +124,54 @@ data "google_iam_policy" "prow_deployer_sa_workload_identity" {
     ]
   }
 }
-// Authoritative iam-policy: replaces any existing policy attached to this service_account
 resource "google_service_account_iam_policy" "prow_deployer_sa_iam" {
   service_account_id = google_service_account.prow_deployer_sa.name
   policy_data        = data.google_iam_policy.prow_deployer_sa_workload_identity.policy_data
 }
-// Create GCP SA for jobs that write to gs://k8s-metrics and gs://k8s-project-metrics
+// roles: prow-deployer (there are also some assigned in ensure-main-project.sh)
+resource "google_project_iam_member" "prow_deployer_for_prow_build_trusted" {
+  project = local.project_id
+  role    = "${data.google_organization.org.name}/roles/container.deployer"
+  member  = "serviceAccount:${local.prow_deployer_sa_name}@${local.project_id}.iam.gserviceaccount.com"
+}
+resource "google_project_iam_member" "prow_deployer_for_prow_build" {
+  project = "k8s-infra-prow-build"
+  role    = "${data.google_organization.org.name}/roles/container.deployer"
+  member  = "serviceAccount:${local.prow_deployer_sa_name}@${local.project_id}.iam.gserviceaccount.com"
+}
+
+// workload_identity_service_account: k8s-metrics
 resource "google_service_account" "k8s_metrics_sa" {
   project      = local.project_id
   account_id   = local.k8s_metrics_sa_name
   display_name = local.k8s_metrics_sa_name
 }
-// Allow pods using the build cluster KSA to use the GCP SA via workload identity
 data "google_iam_policy" "k8s_metrics_sa_workload_identity" {
   binding {
     role = "roles/iam.workloadIdentityUser"
-
     members = [
       "serviceAccount:${local.project_id}.svc.id.goog[${local.pod_namespace}/${local.k8s_metrics_sa_name}]",
     ]
   }
 }
-// Authoritative iam-policy: replaces any existing policy attached to this service_account
 resource "google_service_account_iam_policy" "k8s_metrics_sa_iam" {
   service_account_id = google_service_account.k8s_metrics_sa.name
   policy_data        = data.google_iam_policy.k8s_metrics_sa_workload_identity.policy_data
 }
-
-resource "google_project_iam_member" "prow_deployer_for_prow_build_trusted" {
-  project = local.project_id
-  role    = "${data.google_organization.org.name}/roles/container.deployer"
-  member  = "serviceAccount:${local.prow_deployer_sa_name}@${local.project_id}.iam.gserviceaccount.com"
-}
-resource "google_project_iam_member" "prow_deployer_for_prow_build" {
-  project = "k8s-infra-prow-build"
-  role    = "${data.google_organization.org.name}/roles/container.deployer"
-  member  = "serviceAccount:${local.prow_deployer_sa_name}@${local.project_id}.iam.gserviceaccount.com"
+// roles: k8s-metrics
+resource "google_project_iam_member" "k8s_metrics_sa_bigquery_user" {
+  project = var.project_name
```

Comment:
Not sure if we use variables in this project.

Reply:
changed to `locals.project_id`